decode_test.go

```diff
@@ -559,6 +559,17 @@ var unmarshalTests = []struct {
 		"a: []",
 		&struct{ A []int }{[]int{}},
 	},
+
+	// Line number information from parser
+	{
+		"\n\n\n\n\n\n\n\n\n\na: one\nb: two\nc: three\n",
+		&struct {
+			A    string
+			B    string
+			Line int ",linenum"
+			Col  int ",colnum"
+		}{"one", "two", 11, 1},
```

Comment:
It would be the line and column of where in the YAML that object "started". It's not perfectly granular, and only works for objects, so I understand completely if this is too hacky for you to want to add to go-yaml.  My use case is I have a file that has an array of several thousand objects. So when I find an error, I can report on what line that object began.